src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -0,0 +1,215 @@
+#include "remote_control/resource_allocation_manager_impl.h"
+#include "application_manager/application.h"
+#include "application_manager/message_helper.h"
+#include "remote_control/rc_module_constants.h"
+#include "json/json.h"
+#include "utils/helpers.h"
+#include "utils/make_shared.h"
+#include "remote_control/message_helper.h"
+
+namespace remote_control {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RemoteControlModule")
+
+ResourceAllocationManagerImpl::ResourceAllocationManagerImpl(
+    RemotePluginInterface& rc_plugin)
+    : current_access_mode_(hmi_apis::Common_RCAccessMode::AUTO_ALLOW)
+    , rc_plugin_(rc_plugin) {}
+
+ResourceAllocationManagerImpl::~ResourceAllocationManagerImpl() {}
+
+AcquireResult::eType ResourceAllocationManagerImpl::AcquireResource(
+    const std::string& module_type, const uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const application_manager::ApplicationSharedPtr acquiring_app =
+      rc_plugin_.service()->GetApplication(app_id);
+  if (!acquiring_app) {
+    LOG4CXX_WARN(logger_, "App with app_id: " << app_id << "does not exist!");
+    return AcquireResult::IN_USE;
```

Comment:
Why would the app not existing return an IN_USE result?

Reply:
@JackLivio we have no separate result code for missed application, and I think that it is redundant. 
The mission of AcquireResource function is only to acquire and return success if acquired or in use if not acquired. Setting incorrect app_id it is runtime error, to deal with it this function should prevent acquiring resource for invalid app_id, and return IN_USE result code.

Reply:
@LuxoftAKutsan I think that IN_USE is misleading for this case. Do you think that REJECTED would make more sense?

Reply:
@JackLivio  not that it is not result code to mobile. It is AcquireResult. And there is no REJECTED result code. And I would prefer not to add it for handling case that is not regular ( not that missed application in scope of this function it is probably runtime error)